kernel/src/sched.rs

```diff
@@ -653,6 +771,14 @@ impl Kernel {
                 }
             }
         }
+        let time_executed_us = timeslice_us.map_or(None, |timeslice| {
+            // min is to protect from wrapping
+            Some(u32::min(
+                timeslice - scheduler_timer.get_remaining_us(),
+                timeslice,
```

Comment:
I had trouble reasoning about when this could possibly wrap, however one scenario I came up with was a coarse-grained timer, in which case the `remaining_us` might be greater than the timeslice if the process executed for a very _short_ window of time; but this correction will determine that the _whole_ timeslice was used rather than almost none of it.

I'm not sure if that's the wrapping that's being protected against? If it is, I don't think this is the right logic. In either case, at least a bit more comment explanation here would help I think.

Reply:
I was thinking of the `VirtualSchedulerTimer` when I wrote this -- I am concerned about the case where the timer wraps around after we return from userspace, but the process has yielded, so we return without checking `scheduler_timer.has_expired()` again. For the virtual timer, the time returned will be very large in this case, so I wanted to coerce any very large values to 0...which this does not do. good catch! I changed it to check `has_expired()` after calling `get_remaining_us()` instead, that is much less brittle.

I also updated the documentation of `get_remaining_us()` to make it more clear why this is needed.